File: src/main/java/botmanager/frostbalance/menu/MenuResponse.java
package botmanager.frostbalance.menu;

public abstract class MenuResponse {

    String emoji;
    String name;

    MenuResponse(String emoji, String name) {
        this.emoji = emoji;
        this.name = name;
    }

    public void applyReaction() {
        if (isValid()) reactEvent();
    }

    public abstract void reactEvent();

    public abstract boolean isValid();

}

File: src/main/java/botmanager/doomstack/menu/response/MenuResponse.kt
package botmanager.doomstack.menu.response

import botmanager.frostbalance.menu.response.MenuAction

abstract class MenuResponse internal constructor(var emoji: String, name: String): MenuAction(name) {

    fun applyReaction() {
        if (isValid) reactEvent()
    }

    abstract fun reactEvent()
    abstract val isValid: Boolean

}